cmd/lingo/main.go

```diff
@@ -71,11 +71,13 @@ func run() error {
 	var metricsAddr string
 	var probeAddr string
 	var concurrencyPerReplica int
+	var requestHeaderTimeout time.Duration
 
 	flag.StringVar(&metricsAddr, "metrics-bind-address", ":8082", "The address the metric endpoint binds to.")
 	flag.StringVar(&probeAddr, "health-probe-bind-address", ":8081", "The address the probe endpoint binds to.")
 	flag.IntVar(&concurrencyPerReplica, "concurrency", concurrency, "the number of simultaneous requests that can be processed by each replica")
 	flag.IntVar(&scaleDownDelay, "scale-down-delay", scaleDownDelay, "seconds to wait before scaling down")
+	flag.DurationVar(&requestHeaderTimeout, "request-header-timeout", 10*time.Second, "amount of time for the client to send headers before a timeout error will occur")
```

Comment:
I think this would break scale from 0 with gpus? sometimes scale up takes 10+ minutes and a request will just be hanging for 10 minutes until the backend comes up.

Reply:
I would vote for setting it to 30 minutes

Reply:
I am actually not sure how ReadHeaderTimeout behaves. Does the proxy/handler first return headers before returning the full body and will it return headers before the backend is up?

If the answer is yes, then I think current value of 10 seconds is fine. I was reading this to improve my understanding: https://adam-p.ca/blog/2022/01/golang-http-server-timeouts/

Reply:
The read header timeout is configured for the lingo http + stats server only and does not affect the proxy or backend timeouts.
This setting protects against slow clients that do not complete their requests. See [slowloris attack](https://medium.com/a-journey-with-go/go-understand-and-mitigate-slowloris-attack-711c1b1403f6) for more context.